src/pronoun-replacement.js

```diff
@@ -1,61 +1,226 @@
+export function pluralizeVerbs(doc) {
+    const subjAdvPat = "they #Adverb? ";
+
+    // Rule for verbs ending in y, like "fly"
+    const yVerbs = doc
+        .match(subjAdvPat + "[_ies]")
+        .out("array")
+        .map(v => [v, v.slice(0, -3) + "y"]);
+
+    // Rule for verbs ending in o, like "go", or a soft consonant, like "squash"
+    const esVerbs = doc
+        .match(subjAdvPat + "[_es]")
+        .out("array")
+        .map(v => [v, v.slice(0, -2)])
+        .filter(v => v[1].match(/(o|s|ch|sh|x|z)$/));
+
+    // Rule for regular verbs ending in other letters
+    const sVerbs = doc
+        .match(subjAdvPat + "[_s]")
+        .out("array")
+        .map(v => [v, v.slice(0, -1)]);
+
+    // Statement form, e.g., "Yes, she smokes." or "No, she does not smoke."
+    // Note we don't need to worry about capitalization, because the verb
+    // can never lead the sentence.
+    irregularVerbs
+        .concat(yVerbs)
+        .concat(esVerbs)
+        .concat(sVerbs)
+        .forEach(function(pair) {
+            doc.match(subjAdvPat + "[" + pair[0] + "]").replaceWith(pair[1]);
+        });
+
+    // Question form, e.g., "Does she smoke?" or "Will not she go?"
+    // Here we do need to worry about capitalization: "Well, does she smoke?"
+    irregularVerbs.forEach(function(pair) {
+        const matches = doc.match("[" + capitalize(pair[0]) + "] not? they");
+        replaceMatchWithCapitalization(matches, substituteIrregularVerb, "");
+    });
+
+    // "Have not they a book?" -> "Have they not a book?"
+    doc.match("not they").replaceWith("they not");
+
+    return doc.all();
+}
+
+// Replace "he" and "she" only.
+// Expects a tagged document, with contractions expanded.
+export function replaceHeShe(doc) {
+    ["she", "he"].forEach(function(pronoun) {
+        const matches = doc.match(pronoun);
+        replaceMatchWithCapitalization(matches, substitutePronoun, "");
+    });
+    return doc.all();
+}
+
+// Replace possessive adjectives:
+//   "her book" -> "their book"
+//   "his book" -> "their book"
+// Expects a tagged document, not plain text!
+export function replacePossessiveAdjectives(doc) {
+    ["her", "his"].forEach(function(pronoun) {
+        const matches = doc.match(
+            "[" + pronoun + "] #Ordinal? #Adjective? #Noun"
+        );
+        replaceMatchWithCapitalization(matches, substitutePronoun, "_poss_adj");
+    });
+    return doc.all();
+}
+
+// Ensures there is a space after every period.
+// This has been a problem with Compromise output.
+export function spaceAfterPeriod(text) {
+    return text.replace(/\./g, ". ").replace(/ +/g, " ");
+}
+
+// Replace the pronouns in given text.
+
+// Algorithm steps:
+// 0. Deal with punctuation that Compromise doesn't handle as we'd like.
+// 1. Replace compound pronouns.
+// 2. Expand relevant contractions.
+// 3. Replace subject "he" and "she" with "they"
+// 4. Pluralize verbs associated with "they"
```

Comment:
If I correctly understand this step, it seems like verbs are correctly conjugated to match new instances of the word "they". I think it would only count as pluralization with the plural "they".